client/src/app/appCtrl.js

```diff
@@ -46,36 +60,127 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
     slidingWindow = false;
     $scope.$broadcast("fileUploadChange");
     $scope.view.loading = true;
-    // we do a quick test here to see if the server supports the Range header.
-    // If so, we try to stream. If not, we try to download.
-    $http.head($scope.view.filePath,{'headers' : {'Range' : 'bytes=0-32'}}).then(function(response){
-      if(response.status === 206) {
-        // now we check to see how big the file is
-        $http.head($scope.view.filePath).then(function(response){
-          var contentLength = response.headers('Content-Length');
-          if (contentLength > appConfig.MAX_FILE_SIZE) {
-            slidingWindow = true;
-            $scope.view.windowing.show = true;
-          }
-          streamRemoteFile($scope.view.filePath);
-        });
-      } else {
-        downloadFile($scope.view.filePath);
-      }
-    }, function() {
-      downloadFile($scope.view.filePath);
-    });
+    $scope.view.file.size = getFileSize($scope.view.filePath);
```

Comment:
I can work on it. In working with web apps like this, is that you have to manage a lot of asynchronous activity. Angular's "promises" handles this for us. The trick is to make sure that any function that depends on an asynchronous call returns a promise, no matter what. That way, anyone calling the function will know what to do with what is returned. Promises can chained together - so you can chain together multiple functions that all return promises to each other.

At this point, I am going to recommend that we do not merge this PR, but instead I will make a copy of your `online_polling` branch, fix these issues, and submit a PR to the `online_polling` branch. If that looks good, we can then do a PR to the master branch. What do you think?